### PR TITLE
[TM] Add spec for AnimatedModule

### PR DESCRIPTION
**Pull request by thymikee**
## Summary

Part of #24875. Added `strict-local` to the NativeModuleHelper btw.

## Changelog

[General] [Added] - TM add spec for AnimatedModule

## Test Plan

Flow green.

### CODE REVIEW

**Comment by uqmessias** [comment]
I was working on that, but I'm glad to see it's finished \o/

**Comment by thymikee** [comment]
@uqmessias so it seems like you're perfectly eligible to help reviewing this 😊

File: Libraries/Animated/src/NativeAnimatedModule.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    eventName: string,
+    eventMapping: EventMapping,
+  ) => void;
+  +removeAnimatedEventFromView: (
+    viewTag: ?number,
+    eventName: string,
+    animatedNodeTag: ?number,
+  ) => void;
+
+  // EventEmitter
```
Comment (uqmessias) [inline Libraries/Animated/src/NativeAnimatedModule.js]: Please, update it to ` Events`

Comment (thymikee) [reply Libraries/Animated/src/NativeAnimatedModule.js]: updated, thanks!

File: Libraries/Animated/src/NativeAnimatedHelper.js [hunk 2]
```diff
@@ -61,7 +58,7 @@ const API = {
   startAnimatingNode: function(
     animationId: ?number,
     nodeTag: ?number,
-    config: Object,
+    config: AnimatinigNodeConfig,
```
Comment (radex) [inline Libraries/Animated/src/NativeAnimatedHelper.js]: typo!

Comment (thymikee) [reply Libraries/Animated/src/NativeAnimatedHelper.js]: Good eye! Thanks

File: Libraries/Animated/src/NativeAnimatedModule.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import * as TurboModuleRegistry from 'TurboModuleRegistry';
+import type {EndCallback} from './animations/Animation';
+
+export type EventMapping = {|
+  nativeEventPath: Array<string>,
+  animatedValueTag: ?number,
+|};
+
+export type AnimatedNodeConfig = {|
+  type?:
```
Comment (fkgozali) [inline Libraries/Animated/src/NativeAnimatedModule.js]: should this be required? (no ?)

File: Libraries/Animated/src/NativeAnimatedModule.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    | 'division'
+    | 'multiplication'
+    | 'modulus'
+    | 'subtraction'
+    | 'transform'
+    | 'tracking',
+|};
+
+export type AnimatingNodeConfig = {|
+  type?: 'frames' | 'spring' | 'decay',
```
Comment (fkgozali) [inline Libraries/Animated/src/NativeAnimatedModule.js]: this too should be required?

File: Libraries/Animated/src/NativeAnimatedModule.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  +createAnimatedNode: (tag: ?number, config: AnimatedNodeConfig) => void;
+  +startListeningToAnimatedNodeValue: (tag: ?number) => void;
+  +stopListeningToAnimatedNodeValue: (tag: ?number) => void;
+  +connectAnimatedNodes: (parentTag: ?number, childTag: ?number) => void;
+  +disconnectAnimatedNodes: (parentTag: ?number, childTag: ?number) => void;
+  +startAnimatingNode: (
+    animationId: ?number,
+    nodeTag: ?number,
+    config: AnimatingNodeConfig,
+    endCallback: EndCallback,
```
Comment (fkgozali) [inline Libraries/Animated/src/NativeAnimatedModule.js]: type imported from other files doesn't work well with our codegen atm, so just inline the type here or copy the type over in this file

**Comment by thymikee** [comment]
@fkgozali addressed the feedback, thanks!

File: Libraries/Animated/src/NativeAnimatedModule.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+type EndCallback = (result: EndResult) => void;
+
+export type EventMapping = {|
+  nativeEventPath: Array<string>,
+  animatedValueTag: ?number,
+|};
+
+export type AnimatedNodeConfig = {|
+  type:
+    | 'style'
```
Comment (fkgozali) [inline Libraries/Animated/src/NativeAnimatedModule.js]: Hmm actually we don't yet support enums, so I'll just update this to string when I import to FB, for now... Sorry for the confusion. cc @RSNara

**Pull request closed without merging**